### DISCUSSION TITLE
Integration card with muliple devices is dificult to use

### DISCUSSION THREAD

**Discussion by tomlut** [home-assistant/frontend#10375] [Configuration] [answered]
When many multiple devices exist for an integration the small scrollable widow is difficult to use. e.g. there are 56 device in this list yet only three are visible at any one time:

![Screenshot 2021-10-23 at 18-24-29 Configuration - Home Assistant](https://user-images.githubusercontent.com/10679300/138546927-91423136-a542-43ae-9d9e-d18b7f014519.png)

**Comment by wolph** [comment]
I can second this issue. I have multiple integrations with this issue. The silly thing is that it's perfectly usable on mobile devices where all items are shown:

![image](https://user-images.githubusercontent.com/270571/207293450-70f99f38-abce-4685-b143-e2611a9bd99f.png)

**Comment by ildar170975** [comment]
May be my words will be unproductive - but this is an example of bad consequences of moving from yaml settings to UI.
As for the problem - may be we should open a large popup where it could be easier to define settings?

**Reply by wolph** [reply]
> May be my words will be unproductive - but this is an example of bad consequences of moving from yaml settings to UI.

That's not the issue in this case I think. Even if the settings for this would be in the yaml, it would still be a terrible interface to look at the available devices and values. 

> As for the problem - may be we should open a large popup where it could be easier to define settings?
That would certainly make it better, but I would opt for larger integration cards in any case. Perhaps a split between grouped and non-grouped integrations would be useful in that case.

I.e. the difference between one of these:
![image](https://user-images.githubusercontent.com/270571/207307048-39dae9cd-b98f-4b41-84bd-c6b14c1dc00e.png)

And these:
![image](https://user-images.githubusercontent.com/270571/207307080-95abfd8b-983a-4aab-8f4b-219cee7e86a2.png)

Although I wouldn't mind having bigger integration cards in all cases. I've got about 50 integrations and it's barely taking up 1/3 of the available screen space

**Reply by ildar170975** [reply]
> That's not the issue in this case I think. Even if the settings for this would be in the yaml, it would still be a terrible interface to look at the available devices and values.

I meant a different case - when the integration is setup in yaml only, no UI even for displaying.
For instance: netdata platform is defined in yaml only, it is quite enough.

> Although I wouldn't mind having bigger integration cards in all cases. I've got about 50 integrations and it's barely taking up 1/3 of the available screen space

My 1920x1200 monitor shows 6x5=30 tiles.
Big tiles may be too redundant for integrations like "Sun".

My proposal was:
1. Show small (like currently) tiles.
2. If it is needed for some tile - show an icon (or a label) on a tile; selecting this element will display a big popup for defining (or displaying) large amount of data.

**Reply by wolph** [reply]
> I meant a different case - when the integration is setup in yaml only, no UI even for displaying. For instance: netdata platform is defined in yaml only, it is quite enough.

Ah, that makes sense.

Wouldn't that be a bit confusing though? If you have to look for integrations in 2 places?

> My 1920x1200 monitor shows 6x5=30 tiles. Big tiles may be too redundant for integrations like "Sun".

Yes, that's definitely a fair point.

> My proposal was:
> 
> 1. Shall small (like currently) tiles.
> 2. If it is needed for some tile - show an icon (or a label) on a tile; selecting this element will display a big popup for defining (or displaying) large amount of data.

I really like the idea of the popup, but to prevent a load of extra clicks constantly it would still be nice to increase the size for larger integrations a bit.

**Reply by ildar170975** [reply]
> Wouldn't that be a bit confusing though? If you have to look for integrations in 2 places?

Not at all. For me )))
I prefer to have ALL integration in yaml - but now it is not possible for many integrations. Keeping settings as a yaml code gives plenty of opportunities.
Sorry for an off-topic. The question "yaml vs UI" should not be discussed here...

**Reply by ildar170975** [reply]
> increase the size for larger integrations a bit.

To maintain 50 devices it will be still not enough.
I propose to use a similar approach as for more-info popup: double-clicking on some element enlarges the current tile.

**Comment by Cl3tus** [comment]
Still hoping to get this feature in here!

**Reply by spicedreams** [reply]
Me too. The only integration I use is Shelly, but I have a couple of dozen devices. The UI makes them extremely difficult to use.

**Comment by Misiu** [comment]
Ref: https://github.com/home-assistant/frontend/discussions/16076

as mentioned there, we can use the search on top of the page to find specific devices/integrations:
![image](https://user-images.githubusercontent.com/1741838/233285356-d1e63537-7ccc-4922-9fd1-7d2254b355a8.png)

**Reply by ildar170975** [reply]
Using a "search" is not always convenient, a user need to know WHAT to search.

**Reply by Cl3tus** [reply]
Exactly, i'm a "GUI" guy, I'd like to check and scroll. I sometimes don't have it in my fingers what i'm searching for.

**Reply by wolph** [reply]
My use-case is that I want to check all/multiple devices of the same integration and in some cases those are autogenerated so I don't have a clue what the name would be. It would be much more convenient if it would just show a list (perhaps even in a new screen) with only those devices.

Another thing that would help a lot if clicking with the middle mouse button would immediately open the page for the device in a new tab, instead of having to click on the device, click on the `1 device` link (it's always 1) and click back. That's a lot more tedious.

**Reply by ildar170975** [reply]
Also, there is one more use-case.
I have an integration XYZ, which has 20 entries, I may only see ~5 of them.
And sometimes SOME of these 20 entries are not setup properly after HA restart - I need to scroll the list and manually reload these entries.
Using "Search" here is not helpful at all.

**Reply by Misiu** [reply]
I totally agree, I also would like to have an easy way of displaying all the devices in integration (or at least a larger number of devices). I suggested search as a temporary solution.
I think that card size should vary by the number of devices, if there are fewer than 5 show them as they are if there are more make the card higher. Sadly I'm not a CSS expert, I've tried making something, but it doesn't look good in all cases (as shown in the linked issue)

**Comment by Misiu** [comment]
I propose this layout: https://codepen.io/Misiu/pen/YzJZWeB
Can someone with a Mac try this on Safari?
Idea is to size cards based on the number of items in a card.

**Comment by wolph** [comment]
Seems to work for me on safari and chrome on an ipad at least

**Comment by Misiu** [comment]
I've created PR with the proposed change: https://github.com/home-assistant/frontend/pull/16314.
Upvotes are welcome :)

**Comment by matthiasdebaat** [comment, marked as answer]
As a first iteration solution, we want to make integration grouping consistent by using the same `6 devices` and `29 entities` design as on other cards. Clicking on devices will take you to the filtered devices page and takes the same amount of clicks. This was made possible because we recently added hubs, devices and services to the [manifest](https://developers.home-assistant.io/blog/2022/10/24/hubs-devices-services-virtual-integrations/ ).

![CleanShot_2023-05-08_at_08 37 00](https://user-images.githubusercontent.com/20203284/237062491-df523f33-4d91-4061-abae-f6c2710bb5f2.png)

Don't have an ETA for this fix.

**Reply by bramkragten** [reply]
https://github.com/home-assistant/frontend/pull/16640

**Comment by tomlut** [comment]
Closed due to being addressed with multiple frontend improvements.
